lib/pacemaker/pcmk_sched_clone.c

```diff
@@ -658,14 +658,12 @@ pcmk__clone_allocate(pe_resource_t *rsc, pe_node_t *prefer,
     for (GListPtr gIter = rsc->rsc_cons_lhs; gIter != NULL; gIter = gIter->next) {
         pcmk__colocation_t *constraint = (pcmk__colocation_t *) gIter->data;
 
-        if (constraint->score == 0) {
-            continue;
+        if (pcmk__colocation_applies(rsc, constraint, false)) {
+            rsc->allowed_nodes = constraint->rsc_lh->cmds->merge_weights(constraint->rsc_lh,
+                    rsc->id, rsc->allowed_nodes, constraint->node_attribute,
+                    constraint->score / (float) INFINITY,
+                    pe_weights_rollback|pe_weights_positive);
```

Comment:
Strictly speaking, this doesn't work identically to the previous code.  This will now skip setting rsc->allowed_nodes due to the promoted_only check that's been added to pcmk__colocation_applies.  That is the case for all but the last of the similar changes in this patch.  However, I think that is probably what you intend.

I see no other differences.

Reply:
When promoted_only is false, that check has no effect, so the behavior is actually the same in this call